xilem/src/view/board/style_modifier.rs

```diff
@@ -0,0 +1,342 @@
+        transform(self, affine)
+    }
+
+    fn fill(self, brush: impl Into<Brush>) -> Fill<Self, State, Action> {
+        fill(self, brush)
+    }
+
+    fn stroke(self, brush: impl Into<Brush>, style: kurbo::Stroke) -> Stroke<Self, State, Action> {
+        stroke(self, brush, style)
+    }
+
+    fn into_any_board(self) -> AnyBoardChild<State, Action>
+    where
+        Self: AnyView<State, Action, ViewCtx, Pod<KurboShape>> + Send + Sync + 'static,
+    {
+        AnyBoardChild::Graphics(Box::new(self))
+    }
+}
+
+impl<State: 'static, Action: 'static> GraphicsExt<State, Action> for kurbo::PathSeg {}
+impl<State: 'static, Action: 'static> GraphicsExt<State, Action> for kurbo::Arc {}
+impl<State: 'static, Action: 'static> GraphicsExt<State, Action> for kurbo::BezPath {}
+impl<State: 'static, Action: 'static> GraphicsExt<State, Action> for kurbo::Circle {}
+impl<State: 'static, Action: 'static> GraphicsExt<State, Action> for kurbo::CircleSegment {}
+impl<State: 'static, Action: 'static> GraphicsExt<State, Action> for kurbo::CubicBez {}
+impl<State: 'static, Action: 'static> GraphicsExt<State, Action> for kurbo::Ellipse {}
+impl<State: 'static, Action: 'static> GraphicsExt<State, Action> for kurbo::Line {}
+impl<State: 'static, Action: 'static> GraphicsExt<State, Action> for kurbo::QuadBez {}
+impl<State: 'static, Action: 'static> GraphicsExt<State, Action> for kurbo::Rect {}
+impl<State: 'static, Action: 'static> GraphicsExt<State, Action> for kurbo::RoundedRect {}
+
+impl<State: 'static, Action: 'static, V: GraphicsView<State, Action>> GraphicsExt<State, Action>
+    for Transform<V, State, Action>
+{
+}
+impl<State: 'static, Action: 'static, V: GraphicsView<State, Action>> GraphicsExt<State, Action>
+    for Fill<V, State, Action>
+{
+}
+impl<State: 'static, Action: 'static, V: GraphicsView<State, Action>> GraphicsExt<State, Action>
+    for Stroke<V, State, Action>
+{
+}
+
+impl<V, State, Action> ViewMarker for Transform<V, State, Action> {}
+impl<State, Action, V> View<State, Action, ViewCtx> for Transform<V, State, Action>
+where
+    State: 'static,
+    Action: 'static,
+    V: GraphicsView<State, Action>,
+{
+    type ViewState = V::ViewState;
+    type Element = Pod<KurboShape>;
+
+    fn build(&self, ctx: &mut ViewCtx) -> (Self::Element, Self::ViewState) {
+        let (mut element, state) = self.child.build(ctx);
+        element
+            .inner
+            .as_mut()
+            .unwrap()
+            .set_transform(self.transform);
+        (element, state)
+    }
+
+    fn rebuild<'el>(
+        &self,
+        prev: &Self,
+        child_state: &mut Self::ViewState,
+        ctx: &mut ViewCtx,
+        element: Mut<'el, Self::Element>,
+    ) -> Mut<'el, Self::Element> {
+        let mut element = self.child.rebuild(&prev.child, child_state, ctx, element);
+        if self.transform != prev.transform {
+            element.set_transform(self.transform);
```

Comment:
I believe we want to check if the child views have changed the transform (and similar for other properties) as well, and overwrite it in case.
That's one reason why there's the rather complicated modifier logic in xilem_web so that this does not lead to surprising results (i.e. the most outer modifier always overwrites previous modifiers of the same type).

I think the simplest way is just something like:

```suggestion
        if self.transform != element.transform {
            element.set_transform(self.transform);
```
But that may have the disadvantage of producing unnecessary repaints, when there's other modifiers that transform that property.

We could also use dirty flags etc. to avoid that issue, something like `if self.transform != prev.transform || element.transform_dirty() {}`, or something similar.

Reply:
How can the properties be changed other than by the parent?

Reply:
Consider the following:

```rust
Rect::new().transform(inner).transform(outer)
```

when `inner` changes, the new value of this will be the new element value unless `outer` changes as well at the same time, while I would expect, that `outer` overwrites this value regardless.

Reply:
Hmm this is definitely not trivial to get right (i.e. idiomatic, robust and as little extra-code as possible).

[This](https://github.com/linebender/xilem/commit/66cedfe9cc28116d94bcafced74caa140b36e2f2) is a dirty/hacky sketch of how to make this work, but I'm not really happy about it, as it requires boolean dirty flags for every property, and is not optimal when reading values (`View::State` can be used though for this issue, to cache previously set values), when considering a potential future API like:

```rust
Rect::new().rotate().translate()
```
I.e. that views read previously set values (in this case the transform from `rotate` in `translate`).

In xilem_web this is less of an issue, as a minimal changeset needs to be computed anyway (accessing DOM via js is expensive), so that justifies the modifier API there, which among other things accomplishes this.

So in that regard a big struct is more convenient (but requires wrapper types). I'm not sure how future-proof that is either, when considering a lot of attributes per widget (i.e. unnecessary memory usage when using the view), and to my thinking, this general pattern (composing views with new types based on trait-bounds on `View::Element`) *should* be possible.

@PoignardAzur and @DJMcNab do you have any suggestions here?